crates/red_knot_python_semantic/src/types.rs

```diff
@@ -6993,6 +6993,10 @@ impl<'db> IntersectionType<'db> {
     pub fn iter_positive(&self, db: &'db dyn Db) -> impl Iterator<Item = Type<'db>> {
         self.positive(db).iter().copied()
     }
+
+    pub fn is_single_valued(&self, db: &'db dyn Db) -> bool {
```

Comment:
We shouldn't use `is_single_valued` for this; that has a different meaning already (a type where all inhabitants compare equal to each other). I would probably name this `element_count()` and have it return a `usize` -- the caller can check if it equals `1`.